### PR TITLE
xcodeTarget to also automatically match Tests and WatchKit App

### PR DESCRIPTION
**Pull request by brunobowden**
For an iOS app target of ‘IOS-APP’, this will add linkages for other
iOS targets that start with the same name. Such as ‘IOS-APPTests’,
‘IOS-APP WatchKit App’ and ‘IOS-APP WatchKit Extension’.

- Automatic linkage for Apple Watch apps and extension
- Matches against all targets that start with the same prefix
- Standardize xcodeTarget examples and test to use ‘IOS-APP’

### CODE REVIEW

**Comment by brunobowden** [comment]
@advayDev1, @confile - PTAL, something I needed to make development easier for WatchKit Apps

**Comment by confile** [comment]
What does this pull request?

File: src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcConfig.groovy [hunk 1]
```diff
@@ -565,8 +564,24 @@ class J2objcConfig {
...
+        verifyNoSpaceArgs(nameArgs, args)
+        listArgs.addAll(Arrays.asList(args))
+    }
+
+    // As above but for String[] instead of List<String>
+    @VisibleForTesting
+    static void appendArgs(String[] arrayArgs, String nameArgs, String... args) {
+        verifyNoSpaceArgs(nameArgs, args)
+        for (String arg in args) {
+            arrayArgs += arg
```
Comment (advayDev1) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcConfig.groovy]: does this actually work?  you can modify the arrayArgs reference passed on?

Comment (brunobowden) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcConfig.groovy]: You know, I was wondering the same thing and decided I need to unit test it... having now added the test, it doesn't work. So I've reverted the changes for String[] variables.

**Comment by brunobowden** [comment]
@advayDev1 - it's all fixed up now

@confile - this makes it easier to work with multiple targets. For example, having a WatchKit app as well.

File: src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcConfig.groovy (hunk 1 above)
Comment (advayDev1) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcConfig.groovy]: yeah for other collections this is an .addAll call, and for arrays it is really an assignment and a plus.  one of the annoyances of groovy magic

**Comment by advayDev1** [comment]
LGTM

**Comment by confile** [comment]
LGTM

**Pull request merged**